syft/pkg/cataloger/redhat/cataloger.go

```diff
@@ -20,7 +22,23 @@ func NewDBCataloger() pkg.Cataloger {
 
 	return generic.NewCataloger("rpm-db-cataloger").
 		WithParserByGlobs(parseRpmDB, pkg.RpmDBGlob).
-		WithParserByGlobs(parseRpmManifest, pkg.RpmManifestGlob)
+		WithParserByGlobs(parseRpmManifest, pkg.RpmManifestGlob).
+		WithProcessors(dependency.Processor(dbEntryDependencySpecifier), denySelfReferences)
+}
+
+func denySelfReferences(pkgs []pkg.Package, rels []artifact.Relationship, err error) ([]pkg.Package, []artifact.Relationship, error) {
+	// it can be common for dependency evidence to be self-referential (e.g. bash depends on bash), which is not useful
+	// for the dependency graph, thus we remove these cases
+	for i := 0; i < len(rels); i++ {
```

Comment:
I see that we're decrementing `i` to account for the filter. Are we doing the loop this way to avoid doubling the relationship allocations?

Reply:
exactly -- if there are no self referential package relationships then we aren't allocating another slice